src/core/qgsvectorfilewriter.cpp

```diff
@@ -2749,21 +2750,27 @@ QString QgsVectorFileWriter::driverForExtension( const QString &extension )
   QString ext = extension.trimmed();
   if ( ext.isEmpty() )
     return QString();
-  QgsApplication::registerOgrDrivers();
-  int const drvCount = OGRGetDriverCount();
+
+  if ( ext.startsWith( '.' ) )
+    ext.remove( 0, 1 );
+
+  GDALAllRegister();
```

Comment:
Doesn't seem to be. It's called when the GDAL provider is constructed, but that's not necessarily going to happen at application startup. GDALAllRegister() is called lots of times throughout the codebase before GDAL driver metadata is used.